thirdparty/scripts/build_arrow.sh

```diff
@@ -88,7 +88,8 @@ if [[ ! -d $TP_DIR/../python/ray/pyarrow_files/pyarrow || \
         -DARROW_PYTHON=on \
         -DARROW_PLASMA=on \
         -DPLASMA_PYTHON=on \
-        -DARROW_JEMALLOC=off \
+        -DARROW_JEMALLOC=on \
```

Comment:
Probably won't change anything, but we should test to see if this has any performance implications.